kustomization.yaml.tpl

```diff
@@ -9,8 +9,8 @@ patches:
     kind: ServiceAccount
     metadata:
       name: kustomize-controller
-      annotations:
-        eks.amazonaws.com/role-arn: ${irsa_role_arn}
+      annotations: ${service_account_annotations}
```

Comment:
II'm not sure if this works in multi-cloud (I mean the annotation-free approach) because Azure Workload Identity is closer to AWS IRSA than AWS Pod Identity and requires some annotations.

Reply:
its looks like Azure has deprecated pod identity and created their own workload identity which require more annotations on service accounts. https://github.com/Azure/aad-pod-identity

Reply:
So I think for Azure, workload identity actually does require annotations and labels to be set on SAs unfortunately, judging from the [docs](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview?tabs=dotnet#service-account-labels-and-annotations). I may be wrong though, and I'd be happy if this wasn't needed